### PR TITLE
Support for Trace Analytics

### PR DESCRIPTION
**Pull request by sarahzinger**
**What this PR does / why we need it**:
Adds support for [Trace Analytics](https://opensearch.org/docs/2.6/observing-your-data/trace/index/):

Screenshot 2023-04-19 at 3 14 31 PM

As part of this work we also redesigned the query editor's informational architecture. It is currently a bit confusing that queries like "raw data" and "logs" are under a metric field and we didn't want to continue that pattern with traces. In the future we hope to move out non-metric related queries from out of the metric dropdown, however to minimize changes we haven't done so yet.

**Which issue(s) this PR fixes**:
This pr contains merges from several other prs that have captured the individual issues.

### CODE REVIEW

File: devenv/README.md [hunk 1]
```diff
@@ -25,4 +25,23 @@ URL: https://localhost:9200
...
+
+
+### How to set up local development environment to work with traces in Open Search
+
+Open Search provides [sample apps](https://opensearch.org/docs/latest/observing-your-data/trace/getting-started/) that are instrumented and can generate trace data out of the box. One of them is  the Jaeger HotROD demo.
+
+#### To set it up and add it as a Grafana datasource:
+
+1. Clone the Data [Prepper project](https://github.com/opensearch-project/data-prepper)
+2. Change the line `image:jaegertracing/example-hotrod:latest` to `image: jaegertracing/example-hotrod:1.41.0`
```
Comment (katebrenner) [inline devenv/README.md:38]:
```suggestion
2. Change the line `image: jaegertracing/example-hotrod:latest` to `image: jaegertracing/example-hotrod:1.41.0` in docker-compose.yml
```

File: devenv/README.md [hunk 2]
```diff
@@ -25,4 +25,23 @@ URL: https://localhost:9200
...
+3. Go to the `examples/jaeger-hotrod` directory and start the containers with `docker compose up`
+
+- The app that generates the traces is at [:8000](http://localhost:8080). Clicking on the buttons in the app generates test traces.
+- The Opensearch Dashboards is at [:5601](http://localhost:5601/app/observability-dashboards#/trace_analytics/home)(Passwords are always admin:admin)
+- The Opensearch database is running at https://localhost:9200
+
+In order to view generated traces in Grafana, add a new datasource, enter https://localhost:9200 as the url. Set “Skip TLS Verify” to true. The auth details are admin:admin.
+After following the steps for querying traces from ../README.md, traces from the sample app should be displayed in the panel. 
+
```
Comment (katebrenner) [inline devenv/README.md:47]: These detailed instructions are 🥇

**Review by katebrenner — APPROVED** [review]
LGTM 👍

File: src/types.ts [hunk 3]
```diff
@@ -95,3 +96,79 @@ export enum Flavor {
   Elasticsearch = 'elasticsearch',
   OpenSearch = 'opensearch',
 }
+
+// TODO add raw data and logs
```
Comment (iwysiu) [inline src/types.ts:100]: I assume this is a not in this commit todo?

File: src/types.ts [hunk 4]
```diff
@@ -95,3 +96,79 @@ export enum Flavor {
...
+export enum LuceneQueryType {
+  Traces = 'Traces',
+  Metric = 'Metric',
+}
+
+export type LuceneQueryObj = {
+  size: number;
+  query: DSLQuery;
+  // TODO we could type this further, default trace queries are very hard coded right now and do not change much.
```
Comment (iwysiu) [inline src/types.ts:109]: If we merge this we should create a follow up issue in the backlog for this

**Review by iwysiu — APPROVED** [review]
Nice! I noticed the buttons slightly don't line up:
Screenshot 2023-04-19 at 4 09 02 PM

and I don't know if we found out if we could remove it or not, but the inoperable `Logs for this span` button is still there
Screenshot 2023-04-19 at 4 16 38 PM

besides that, just some housekeeping and teeny style nits

File: src/utils.ts [hunk 5]
```diff
@@ -60,3 +62,69 @@ export async function sha256(string) {
   const hashHex = hashArray.map(bytes => bytes.toString(16).padStart(2, '0')).join('');
   return hashHex;
 }
```
[the remark this answers is not in the document]
Comment (sarahzinger) [reply src/utils.ts:64]: I would think our linter would yell at us if this was a thing? hmm

File: src/types.ts (hunk 3 above)
Comment (sarahzinger) [reply src/types.ts:100]: made a ticket for this and put it in our backlog https://github.com/grafana/opensearch-datasource/issues/149

File: src/types.ts (hunk 4 above)
Comment (sarahzinger) [reply src/types.ts]: good catch! just updated so we don't have this unknown anymore

**Pull request merged**